### PR TITLE
Update the guidelines.

### PR DESCRIPTION
**Pull request by ewels**
Changes are based on discussion from the nf-core 2022 core team retreat.

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1257"></a>

### CODE REVIEW

File: markdown/developers/guidelines/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+## If the guidelines don't fit
+
+We appreciate that the above guidelines are relatively rigid and may not be for everyone.
+If that's the case, there is still a lot of ways that you can get involved with nf-core!
+
+### Non nf-core pipelines
+
+If a pipeline does not conform to nf-core guidelines, it can of course exist outside of nf-core.
+You are verly wecome to the helper tools and collaborate on modules / subworkflows / ideas.
```
Comment (apeltzer) [inline markdown/developers/guidelines/index.md]:
```suggestion
You are very wecome to the helper tools and collaborate on modules / subworkflows / ideas.
```

File: markdown/developers/guidelines/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+## If the guidelines don't fit
+
+We appreciate that the above guidelines are relatively rigid and may not be for everyone.
+If that's the case, there is still a lot of ways that you can get involved with nf-core!
+
+### Non nf-core pipelines
+
+If a pipeline does not conform to nf-core guidelines, it can of course exist outside of nf-core.
+You are very wecome to the helper tools and collaborate on modules / subworkflows / ideas.
```
Comment (jfy133) [inline markdown/developers/guidelines/index.md:66]:
```suggestion
You are very welcome to the helper tools and collaborate on modules / subworkflows / ideas.
```

**Review by apeltzer — CHANGES_REQUESTED** [review]

File: markdown/developers/guidelines/requirements/acknowledgements.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+title: Credits and Acknowledgements
+subtitle: Pipelines must properly acknowledge prior work.
+menu:
+  main:
+    weight: 140
+---
+
+Where previous work from other pipelines / projects is used within a pipeline, the original author(s) must be properly acknowledged.
+Some examples and on how you could do that to make sure they feel valued:
```
Comment (apeltzer) [inline markdown/developers/guidelines/requirements/acknowledgements.md:10]:
```suggestion
Some examples on how you could do that to make sure they feel valued:
```

File: markdown/developers/guidelines/index.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+## Introduction
+
+The aim of nf-core is to have standardised best-practice pipelines.
+To ensure this standardisation, we maintain a set of guidelines which all nf-core
+pipelines must adhere to.
+
+> If you're thinking of adding a new pipeline to nf-core, please read the documentation
+> about [adding a new pipeline](adding_pipelines.md).
+
+The following lists an overview of the guidelines. Follow links to dedicated pages for more detail about a given topic.
```
Comment (FriederikeHanssen) [inline markdown/developers/guidelines/index.md:18]:
```suggestion
The following lists an overview of the guidelines. Follow links to dedicated pages for more details about a given topic.
```
? maybe an s missing

File: markdown/developers/guidelines/index.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+---
+
+## Introduction
+
+The aim of nf-core is to have standardised best-practice pipelines.
+To ensure this standardisation, we maintain a set of guidelines which all nf-core
+pipelines must adhere to.
+
+> If you're thinking of adding a new pipeline to nf-core, please read the documentation
+> about [adding a new pipeline](adding_pipelines.md).
```
Comment (FriederikeHanssen) [inline markdown/developers/guidelines/index.md:16]: the usual sentence about approaching us in new-pipelines?

File: markdown/developers/guidelines/index.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+The following lists an overview of the guidelines. Follow links to dedicated pages for more detail about a given topic.
+
+### Ask the community
+
+The instructions above are subject to interpretation and specific scenarios.
```
Comment (FriederikeHanssen) [inline markdown/developers/guidelines/index.md:22]: below?

File: markdown/developers/guidelines/recommendations/cloud_compatible.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  main:
+    weight: 230
+---
+
+Pipelines should have explicit support for running in cloud environments.
+
+Specifically, they should use the automated cloud-compute tests that are available for free through nf-core.
+The [nf-core template](/tools#creating-a-new-workflow) comes with all required code to support this setup.
+
+Where possible, hope that nf-core pipelines can also have benchmarks for pricing and performance when running on cloud environments.
```
Comment (FriederikeHanssen) [inline markdown/developers/guidelines/recommendations/cloud_compatible.md:14]: don't quite get the 'hope'

File: markdown/developers/guidelines/requirements/parameters.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+---
+title: Standardised parameters
+subtitle: Strive to have standardised usage.
+menu:
+  main:
+    weight: 100
+---
+
+Where possible pipeliens should use the same command line option names as other pipelines for comparable options.
```
Comment (jfy133) [inline markdown/developers/guidelines/requirements/parameters.md:9]:
```suggestion
Where possible pipelines should use the same command line option names as other pipelines for comparable options.
```

File: markdown/developers/guidelines/requirements/parameters.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+Where possible pipelines should use the same command line option names as other pipelines for comparable options.
+For example, `--input` and `--genome`.
+
+In addition to the names of paramters, they should ideally work in a similar way.
+For example, `--input` typically takes a `.csv` sample sheet file (but not always, where not appropriate).
+
+> We're planning to build a tool that lists every paramter used by every pipeline, so that you can check for existing parameters with similar names
+> You can track the progress of this feature request here: []()
```
Comment (jfy133) [inline markdown/developers/guidelines/requirements/parameters.md:16]: Missing link

File: markdown/developers/guidelines/requirements/workflow_specificity.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
+---
+title: Workflow specificity
+subtitle: There should only be a single pipeline per data / analysis type.
+menu:
+  main:
+    weight: 20
+---
+
+The nf-core community was founded to allow different groups to collaborate on pipelines instead of reinventing the same workflows in each institute.
```
Comment (jfy133) [inline markdown/developers/guidelines/requirements/workflow_specificity.md:9]: Always ask for advice on #new-pipelines?

File: markdown/developers/guidelines/requirements/parameters.md (hunk 8 above)
Comment (FriederikeHanssen) [inline markdown/developers/guidelines/requirements/parameters.md]:
```suggestion
Where possible pipelines should use the same command line option names as other pipelines for comparable options.
```

**Review by apeltzer — APPROVED** [review]
Just tiny bits.

File: markdown/developers/guidelines/requirements/workflow_specificity.md (hunk 10 above)
Comment (apeltzer) [reply markdown/developers/guidelines/requirements/workflow_specificity.md:9]: Noted down the same thing a bit more down in another section too :-)

File: markdown/developers/guidelines/requirements/acknowledgements.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+- Send them a message via Slack and let them know that you use their work & had to change something to fit your own purpose. If in doubt, check with them to see how they would like to be acknowledged.
+- Check the licence of their code and make sure you obey the rules that this licence imposes (e.g. `CC-BY` means you have to attribute the original creator).
```
Comment (FriederikeHanssen) [inline markdown/developers/guidelines/requirements/acknowledgements.md:13]:
```suggestion
- Check the licence of their code or graphics components and make sure you obey the rules that this licence imposes (e.g. `CC-BY` means you have to attribute the original creator).
```

File: markdown/developers/guidelines/requirements/identity_branding.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+### The nf-core repository should be set as the _head_ repo
+
+Your personal / organisation repos should show on GitHub as being forked from nf-core, not the other way around.
+This is so that it's clear where the primary development location is.
+It also means that any pull-requests that are created will automatically select the nf-core repository as the target.
+
+When new pipelines are added to nf-core, please _move_ the repository to nf-core instead of forking it.
+
+If you have already forked your pipeline to nf-core, you can email GitHub support and request that they change the head repository.
```
[the remark this answers is not in the document]
Comment (ewels) [reply markdown/developers/guidelines/requirements/identity_branding.md]: Added 👍🏻

File: markdown/developers/guidelines/index.md (hunk 6 above)
Comment (ewels) [reply markdown/developers/guidelines/index.md]: Yup 👍🏻  Copy paste error..

File: markdown/developers/guidelines/recommendations/cloud_compatible.md (hunk 7 above)
Comment (ewels) [reply markdown/developers/guidelines/recommendations/cloud_compatible.md]: Was poorly written. Meant that we hope that people will do benchmarks. Rephrased 👍🏻

File: markdown/developers/guidelines/requirements/docs.md [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+subtitle: Pipeline documentation must be hosted on the nf-core website
+menu:
+  main:
+    weight: 60
+---
+
+All documentation must be bundled with the pipeline code in the main repository, within a directory called `docs`.
+
+Documentation must _only_ be hosted on the nf-core website and GitHub.
+Hosting the documentation at a second location (such as custom readthedocs website, or GitHub pages etc) is not allowed.
```
[the remark this answers is not in the document]
Comment (ewels) [reply markdown/developers/guidelines/requirements/docs.md]: Nice, added 👍🏻

File: markdown/developers/guidelines/requirements/parameters.md (hunk 9 above)
Comment (ewels) [reply markdown/developers/guidelines/requirements/parameters.md]: Ah yeah, well spotted 👍🏻  Chris was writing it as I typed this, but forgot to go back.

File: markdown/developers/guidelines/requirements/workflow_specificity.md (hunk 10 above)
Comment (ewels) [reply markdown/developers/guidelines/requirements/workflow_specificity.md]: Added 👍🏻

**Pull request closed without merging**